Modules/_pickle.c

```diff
@@ -6297,14 +6297,13 @@ load_mark(UnpicklerObject *self)
             return -1;
         }
```

Comment:
Seems the check above can be removed. `alloc > (PY_SSIZE_T_MAX / sizeof(Py_ssize_t))` is duplicated in `PyMem_RESIZE()`, and `alloc <= ((size_t)self->num_marks + 1)` is not possible if `sizeof(Py_ssize_t)` > 2.

Reply:
Agree. This condition can be written as `self->num_marks >= self->marks_size` or `self->num_marks == self->marks_size`.

Reply:
@serhiy-storchaka 
I'd prefer to make these change as separate commits/RPs as they are not related to this bpo issue.

Reply:
PR for upsizing issue https://github.com/python/cpython/pull/8860.